### ISSUE TITLE
exp: compare `cosine` to `wsd`

### ISSUE THREAD

**Issue by eric-czech** [Open-Athena/MarinFold#30]
Compare the WSD-style linear-decay LR schedule used by the exp11 data-mix sweeps against a Levanter-default cosine (full-decay) schedule at 100M on the **m11** size-proportional protein-docs mixture (Open-Athena/MarinFold#11). Use LR from tuning w/ cosine schedule to advantage it.

### Setup

- **Model:** 100M Llama (h=768, l=12, dff=3072, heads=12, kv=4) — same config as [`train_protein_100m_distance_masked.py`][model].
- **Data:** m11 — H/M/L train splits at weights 0.31 / 0.26 / 0.43 (proportional to per-split sizes), loss-masked to distance-bin tokens.
- **Optimizer:** batch=128, seq=8192, 4104 steps (~4.3B tokens), warmup=0.1, LR=9.33e-4 + β₂=0.98 adapted from [`train_protein_1_5b_distance_masked.py`][lineage] via the sqrt-batch / inverse-hidden scaling.
- **Variants:**
    - `wsd` — `lr_schedule="linear"`, `decay=0.2` (linear ramp down over the trailing 0.2 fraction of steps).
    - `cosine` — `lr_schedule=None`, `decay=None` ⇒ `AdamConfig` defaults: cosine schedule, full decay. Matches the recipe in [`train_protein_1_5b_distance_masked.py`][lineage].
- **Eval:** cd-val + H/M/L (val, test) at 16 batches × 128 examples per split (~2k examples / 16.8M tokens each).

### Code

[`experiments/protein/exp30_lrsch_sweep.py`][sweep] on branch `eac/plm-exp30`.

### Runs

W&B group [`exp30-lrsch`][group] — [wsd][wsd] · [cosine][cosine].

[sweep]: https://github.com/marin-community/marin/blob/860985a1c/experiments/protein/exp30_lrsch_sweep.py
[model]: https://github.com/marin-community/marin/blob/860985a1c/experiments/protein/train_protein_100m_distance_masked.py
[lineage]: https://github.com/marin-community/marin/blob/860985a1c/experiments/protein/train_protein_1_5b_distance_masked.py
[group]: https://wandb.ai/eric-czech/marin/groups/exp30-lrsch
[wsd]: https://wandb.ai/eric-czech/marin/runs/prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de
[cosine]: https://wandb.ai/eric-czech/marin/runs/prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf

**Comment by eric-czech** [comment]
## Results

![WSD vs cosine — val loss by trial @ step 4103](https://gist.githubusercontent.com/eric-czech/0f5b657aa923e5ac2e92df1d75f93386/raw/8edd8424cc78e50dabf12f9a61b7b0d745d68886/wsd_vs_cosine.png)

<details><summary>Data</summary>

```
variant_id,variant_name,metric,value,step_used,ref_step,run_name,run_state
wsd,"WSD (linear, decay=0.2)",cd-val-unmasked,10.504209518432615,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",cd-val,1.454275131225586,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",high-test,1.449126124382019,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",high-val,1.454275131225586,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",low-test,1.4903866052627563,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",low-val,1.5135786533355713,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",medium-test,1.4692249298095703,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",medium-val,1.488054275512695,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
wsd,"WSD (linear, decay=0.2)",train,1.5158812999725342,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-wsd-lr9.3e-4-v1-32c6de,finished
cosine,Cosine (full decay),cd-val-unmasked,10.33173370361328,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),cd-val,1.4561928510665894,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),high-test,1.451451301574707,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),high-val,1.4561928510665894,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),low-test,1.492821455001831,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),low-val,1.5164474248886108,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),medium-test,1.4715559482574463,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),medium-val,1.4905829429626465,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
cosine,Cosine (full decay),train,1.518803596496582,4103,4103,prot-exp11-dm-lrsch-100m-4.303B-m11-cosine-lr9.3e-4-v1-7748bf,finished
```

</details>

WSD edges cosine slightly across every val split.

Raw results: [exp30_lrsch_results.zip](https://github.com/user-attachments/files/28325476/exp30_lrsch_results.zip)
